src/prototype/ScenarioSelect.gd

```diff
@@ -16,4 +16,15 @@ func _ready():
 
 
 func _on_Button_pressed():
-		get_tree().change_scene('res://prototype/Main.tscn')
+	
+	global.current_spawns = {
+		preload("res://prototype/Character/Enemy/Enemy.tscn"): 1,
```

Comment:
Shouldn't this be in `global.gd`? Why externally edit these dictionaries instead of defining them within the file itself?

Reply:
Oh wait, never mind. This needs to be scenario-specific.

Reply:
Global variables are an anti-pattern so we have to be careful. There's no easy way to pass data around to scenes on construction because we use `change_scene`. So unfortunately, global variables are the only decent solution here.

Reply:
Perhaps no easy way, but a way nonetheless. See: https://godotengine.org/qa/24773/how-to-load-and-change-scenes?show=24799#a24799

It is said:
***

The simplest is this:
```gdscript
get_tree().change_scene("res://path/to/scene.tscn")
```
Under the hood, changing scene is as simple as replacing a node in the tree by another:

```gdscript
# Remove the current level
var level = root.get_node("Level")
root.remove_child(level)
level.call_deferred("free")  # CI: queue_free()?

# Add the next level
var next_level_resource = load("res://path/to/scene.tscn)
var next_level = next_level.instance()
# CI: pass parameters here
root.add_child(next_level)
```
root can be any node you want. In the first simplest example, root is the root of the whole scene tree. Once you understand how this works, you can apply this anywhere, and you'll realize "changing scene" is a relative concept. You could change everything if you do it on the root, but you can do that to a part of the tree, so that not everything has to change.

pgregory suggested background loading but it's not required to change scene. You might need it only if your scene is too big to the point of needing a loading screen.

***

However, changing the scene through the SceneTree does a bit more than that; it also changes some properties related to the SceneTree.

See: https://github.com/godotengine/godot/blob/master/scene/main/scene_tree.cpp#L1255-L1286

It appears that `_change_scene` is the only method which actually *changes* the scene. `change_scene` and `change_scene_to` just prepare what you give them for `_change_scene`

So, looks like we can get away with something like this:
```gdscript
var new_scene_cls = preload('res://path/to/scene.tscn')
var new_scene = new_scene_cls.instance()
# pass parameters here
get_tree()._change_scene(new_scene)
```

Might be a bit too hacky, using the engine as it was not intended to be used.

Reply:
We don't pursue hacky solutions. A much cheaper, simpler alternative is simply to subclass `Scene` and hard-code both of these values inside each scene instance. I chose the easier approach (global variables).

This is fine as-is for the prototype.